### PR TITLE
Upgrade Infinispan to 14.0.13.Final

### LINKED ISSUE #21564: Upgrade to Infinispan 14.0.13.Final
### Description

This might resolve #21256 but this will need to be tested thoroughly afterwards.

### Motivation

Fix known problems in Infinispan 14.x release

### PR DESCRIPTION
**Pull request by ahus1**
Closes #21564

### CODE REVIEW

**Comment by ahus1** [comment]
Waiting for Infinispan 14.0.12 Docker image to become available as this is needed for the tests: https://quay.io/repository/infinispan/server?tab=tags&tag=latest

**Comment by pruivo** [comment]
@ahus1 My suggestion would be to use `infinispan-bom`. Example: https://github.com/pruivo/keycloak/commit/c495233f6e9ff9b60a7d0cd719f7fefdda35c1be

**Comment by ahus1** [comment]
@pruivo - yes, using the BOM would be a good thing. Still I don't know what else breaks/changes around that. When moving to the BOM, I'd also see some versions set in the "quarkus/runtime/pom.xml" and it would require more thought to clean that up IMHO. 

Let's keep the fingers crossed that a small change is possible for this upgrade.

The most confusing thing is that it was putting the right dependencies in the Quarkus distribution of Keycloak, but used the wrong library in some module's JUnit tests. Definitely something to clean up in a follow-up PR.

**Comment by ahus1** [comment]
Model tests 

```
mvn -B test -Dtest=RoleModelTest -Phot-rod
```

fail with 

>  RoleModelTest.testCompositeRolesPaginationQueries » HotRodClient org.hibernate.search.util.common.SearchException: Exception while searching locally
java.util.concurrent.ExecutionException: org.hibernate.search.util.common.SearchException: HSEARCH000587: Unable to provide the exact total hit count: only a lower-bound approximation is available. This is generally the result of setting query options such as a timeout or the total hit count threshold. Either unset these options, or retrieve the lower-bound hit count approximation through '.total().hitCountLowerBound()'.
org.hibernate.search.util.common.SearchException: HSEARCH000587: Unable to provide the exact total hit count: only a lower-bound approximation is available. This is generally the result of setting query options such as a timeout or the total hit count threshold. Either unset these options, or retrieve the lower-bound hit count approximation through '.total().hitCountLowerBound()'.

due to https://issues.redhat.com/browse/ISPN-15036. 

I'm considering this issue now blocked until another Infinispan micro release. Needing some help from @mhajas / @martin-kanis, possibly @pruivo / @ryanemerson to apply a possible workaround if we decide we don't want to wait.

**Comment by ahus1** [comment]
Waiting for files to appear on Maven central

**Comment by vmuzikar** [comment]
> @pruivo - yes, using the BOM would be a good thing. Still I don't know what else breaks/changes around that. When moving to the BOM, I'd also see some versions set in the "quarkus/runtime/pom.xml" and it would require more thought to clean that up IMHO.
> 
> Let's keep the fingers crossed that a small change is possible for this upgrade.
> 
> The most confusing thing is that it was putting the right dependencies in the Quarkus distribution of Keycloak, but used the wrong library in some module's JUnit tests. Definitely something to clean up in a follow-up PR.

This is actually related to our recent discussion in https://github.com/keycloak/keycloak/issues/21363.

I can imagine there might be some conflicts between deps coming from Quarkus BOM and deps coming from ISPN BOM. Question also is productization. I still do think that ultimately we should align with Quarkus to avoid conflicts but I understand we need to adopt ISPN releases faster.

File: pom.xml [hunk 1]
```diff
@@ -1893,6 +1903,24 @@
         </pluginManagement>
     </build>
 
+    <!-- Enable this only in branches to experiment with snapshot releases from, for example, Infinispan -->
```
Comment (mhajas) [inline pom.xml:1906]: This does not look very nice. Should we rather add this here: https://github.com/keycloak/keycloak/blob/main/maven-settings.xml hidden behind some profile?

**Review by vmuzikar — DISMISSED** [review]
Changes LGTM from cloud native perspective.

**Review by mhajas — COMMENTED** [review]
Thank you @ahus1, the PR looks good, I added one comment to the main pom file.

**Review by mhajas — APPROVED** [review]
Thanks for the update @ahus1

**Pull request merged**